Ignore aarch64 for this test as it's x86 assembly only.  Fixes #7091

ctennis opened the pull request:
fixes #7091 - asm_syntax lint test will not compile on aarch64

changelog: none

rust-highfive commented:
Thanks for the pull request, and welcome! The Rust team is excited to review your changes, and you should hear from @phansch (or someone else) soon.

If any changes to this PR are deemed necessary, please add them as extra commits. This ensures that the reviewer can see what has changed since they last reviewed the code. Due to the way GitHub handles out-of-date commits, this should also make it reasonably obvious what issues have or haven't been addressed. Large or tricky changes may require several passes of review and changes.

Please see [the contribution instructions](https://github.com/rust-lang/rust-clippy/blob/master/CONTRIBUTING.md) for more information.

flip1995 commented:
You have to update the test reference files. You can do this with:

```
cargo uitest
cargo dev bless
```

ctennis commented:
Something is still not quite right here, because when I run `cargo dev bless`, it still wants to put more output into the test - it still seems to be running the test and generating output even though it's being ignored on the arch.  However `cargo uitest` doesn't fail anymore.

ctennis commented:
Or I should say, it seems like it wants to add stderr when it's compiles the test file (and gets the various att_syntax option is only supported on x86) but it doesn't actually run the test to generate any errors.

flip1995 commented:
`cargo dev bless` sometimes tries to use stale files in your `target` dir to update the test files. 

Do I understand correctly, that this version of the PR now passes for you locally on aarch64? If so, please squash your commits and we can merge this as-is.

ctennis commented:
When running "cargo uitest" it passes, and the reason it passes is because it doesn't run the test anymore.

However, it seems like it still compiles the test - and the test compilation phase will want to update the stderr portion with additional error text because of the platform issue:

```
+error: the `att_syntax` option is only supported on x86
+  --> $DIR/asm_syntax.rs:11:9
+   |
+LL |         asm!("", options(att_syntax));
+   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+error: the `att_syntax` option is only supported on x86
+  --> $DIR/asm_syntax.rs:12:9
+   |
+LL |         asm!("", options(nostack, att_syntax));
+   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+error: the `att_syntax` option is only supported on x86
+  --> $DIR/asm_syntax.rs:22:9
+   |
+LL |         asm!("", options(att_syntax));
+   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+error: the `att_syntax` option is only supported on x86
+  --> $DIR/asm_syntax.rs:23:9
+   |
+LL |         asm!("", options(nostack, att_syntax));
+   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

But I can't update that stderr file with this additional output, or it won't match on non aarch64 platforms when the test running is done.

Is there a way somehow I can conditionally skip compilation of this test file as well so it won't want to update that output into the stderr file?  My interpretation of the `//only-x86_64` at the top of the file was it would not compile at all on this platform, but that doesn't seem to be the case.

flip1995 commented:
> When running "cargo uitest" it passes, and the reason it passes is because it doesn't run the test anymore.

Then this should be all good now.

---

I think what's going on is, that before `// only-x86_64` was at the top of the file, the test was run on your machine and produced a reference file in the `target` dir with the additional error messages. Even though the test is now skipped on your machine, the previously produced reference file still exist.

Can you try to run `cargo clean`+`cargo uitest` and then try if it still wants to update the test files?

ctennis commented:
Ok, reset with cargo clean and now back to bad behavior again.

The only way I can make it work is to add `// ignore-aarch64` to the source file. This seems to skip the test running on my platform (though why does it run with `only-x86_64`?).  This makes `cargo uitest` output happy.

But the minute I run `cargo dev bless` it wants to add the compilation errors to the .stderr file:

```
+error: the `att_syntax` option is only supported on x86
+  --> $DIR/asm_syntax.rs:10:9
+   |
+LL |         asm!("", options(att_syntax));
+   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+error: the `att_syntax` option is only supported on x86
+  --> $DIR/asm_syntax.rs:11:9
+   |
+LL |         asm!("", options(nostack, att_syntax));
+   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+error: the `att_syntax` option is only supported on x86
+  --> $DIR/asm_syntax.rs:21:9
+   |
+LL |         asm!("", options(att_syntax));
+   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+error: the `att_syntax` option is only supported on x86
+  --> $DIR/asm_syntax.rs:22:9
+   |
+LL |         asm!("", options(nostack, att_syntax));
+   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
```

So it seems like the comment hints are only impacting running the tests, but not trying to compile them.

ctennis commented:
ok, my latest commit compiles on aarch64 and dev bless has no changes to stderr. However, it will need 'dev blessed' on an x86 platform since the line numbers have changed slightly, the stderr output will be slightly different.

flip1995 commented:
This is a bug in `cargo dev bless` then. Or in compiletest. If the test is ignored, it should not result in the file being updated. cc @phansch @xFrednet do you have any idea what's happening here (compiletest ignoring the test, but cargo dev still updating the reference file)?

---

@ctennis If the test is ignore when you just move the `// only-x86_64` to the top of the file, please only make this change. It doesn't hurt the test suite if a test is wrongly updated. If that happens you can easily revert this with git and don't have to push it.

ctennis commented:
I updated the test file main() to have a `#[cfg(target_arch = "x86_64")]` which fixes the compilation errors for me.  I've used a cross compiled x86_64 rust to rebuild the dev bless output, and squashed the commit. So I think we're in a good place now (aside from the why it compiled in the first place)

flip1995 commented:
@bors r+

Thanks!

bors commented:
:pushpin: Commit 42d070286e993fe671dc4655a545c5c9e3a1feb4 has been approved by `flip1995`

<!-- @bors r=flip1995 42d070286e993fe671dc4655a545c5c9e3a1feb4 -->
<!-- homu: {"type":"Approved","sha":"42d070286e993fe671dc4655a545c5c9e3a1feb4","approver":"flip1995"} -->

bors commented:
:hourglass: Testing commit 42d070286e993fe671dc4655a545c5c9e3a1feb4 with merge 3df6c598a823d316e178fb7e7a4b0ed97fa77aec...
<!-- homu: {"type":"BuildStarted","head_sha":"42d070286e993fe671dc4655a545c5c9e3a1feb4","merge_sha":"3df6c598a823d316e178fb7e7a4b0ed97fa77aec"} -->

bors commented:
:broken_heart: Test failed - [checks-action_remark_test](https://github.com/rust-lang/rust-clippy/runs/2398111743)
<!-- homu: {"type":"BuildFailed","builder_url":"https://github.com/rust-lang/rust-clippy/runs/2398111743","builder_name":"checks-action_remark_test"} -->

flip1995 commented:
@bors retry

bors commented:
:hourglass: Testing commit 42d070286e993fe671dc4655a545c5c9e3a1feb4 with merge 1af0d16cbbe5457718e25e290fb11c5044917662...
<!-- homu: {"type":"BuildStarted","head_sha":"42d070286e993fe671dc4655a545c5c9e3a1feb4","merge_sha":"1af0d16cbbe5457718e25e290fb11c5044917662"} -->

bors commented:
:broken_heart: Test failed - [checks-action_remark_test](https://github.com/rust-lang/rust-clippy/runs/2398200978)
<!-- homu: {"type":"BuildFailed","builder_url":"https://github.com/rust-lang/rust-clippy/runs/2398200978","builder_name":"checks-action_remark_test"} -->

flip1995 commented:
@bors retry

bors commented:
:hourglass: Testing commit 42d070286e993fe671dc4655a545c5c9e3a1feb4 with merge 3254e98dc925c0f86806669e5a955607b762bffe...
<!-- homu: {"type":"BuildStarted","head_sha":"42d070286e993fe671dc4655a545c5c9e3a1feb4","merge_sha":"3254e98dc925c0f86806669e5a955607b762bffe"} -->

flip1995 commented:
@ctennis Nothing for you to do on this PR GitHub Actions is currently having problems: https://www.githubstatus.com/

bors commented:
:broken_heart: Test failed - [checks-action_test](https://github.com/rust-lang/rust-clippy/runs/2398473473)
<!-- homu: {"type":"BuildFailed","builder_url":"https://github.com/rust-lang/rust-clippy/runs/2398473473","builder_name":"checks-action_test"} -->

flip1995 commented:
@bors retry

bors commented:
:hourglass: Testing commit 42d070286e993fe671dc4655a545c5c9e3a1feb4 with merge 926286acfeeec25a423d550083c4e2dacb027e51...
<!-- homu: {"type":"BuildStarted","head_sha":"42d070286e993fe671dc4655a545c5c9e3a1feb4","merge_sha":"926286acfeeec25a423d550083c4e2dacb027e51"} -->

bors commented:
:sunny: Test successful - [checks-action_dev_test](https://github.com/rust-lang/rust-clippy/runs/2398661716), [checks-action_remark_test](https://github.com/rust-lang/rust-clippy/runs/2398653524), [checks-action_test](https://github.com/rust-lang/rust-clippy/runs/2398760092)
Approved by: flip1995
Pushing 926286acfeeec25a423d550083c4e2dacb027e51 to master...
<!-- homu: {"type":"BuildCompleted","approved_by":"flip1995","base_ref":"master","builders":{"checks-action_test":"https://github.com/rust-lang/rust-clippy/runs/2398760092","checks-action_dev_test":"https://github.com/rust-lang/rust-clippy/runs/2398661716","checks-action_remark_test":"https://github.com/rust-lang/rust-clippy/runs/2398653524"},"merge_sha":"926286acfeeec25a423d550083c4e2dacb027e51"} -->
